jigglypuffRL/deeprl/sac/sac.py

```diff
@@ -0,0 +1,388 @@
+        save_interval=5000,
+        layers=(256, 256),
+        tensorboard_log=None,
+        seed=None,
+        render=False,
+        device="cpu",
+        pretrained=False,
+        save_name=None,
+        save_version=None,
+    ):
+
+        self.network_type = network_type
+        self.env = env
+        self.gamma = gamma
+        self.replay_size = replay_size
+        self.batch_size = batch_size
+        self.lr = lr
+        self.alpha = alpha
+        self.polyak = polyak
+        self.entropy_tuning = entropy_tuning
+        self.epochs = epochs
+        self.start_steps = start_steps
+        self.steps_per_epoch = steps_per_epoch
+        self.max_ep_len = max_ep_len
+        self.start_update = start_update
+        self.update_interval = update_interval
+        self.save_interval = save_interval
+        self.layers = layers
+        self.tensorboard_log = tensorboard_log
+        self.seed = seed
+        self.render = render
+        self.save_name = save_name
+        self.save_version = save_version
+        self.save = save_params
+        self.load = load_params
+        self.evaluate = evaluate
+        self.checkpoint = self.__dict__
+        self.pretrained = pretrained
+
+        # Assign device
+        if "cuda" in device and torch.cuda.is_available():
+            self.device = torch.device(device)
+        else:
+            self.device = torch.device("cpu")
+
+        # Assign seed
+        if seed is not None:
+            torch.manual_seed(seed)
+            torch.backends.cudnn.deterministic = True
+            torch.backends.cudnn.benchmark = False
+            np.random.seed(seed)
+            self.env.seed(seed)
+            random.seed(seed)
+
+        # Setup tensorboard writer
+        self.writer = None
+        if self.tensorboard_log is not None:
+            from torch.utils.tensorboard import SummaryWriter
+
+            self.writer = SummaryWriter(log_dir=self.tensorboard_log)
+
+        self.create_model()
+
+    def create_model(self):
+        state_dim = self.env.observation_space.shape[0]
+
+        # initialize models
+        if isinstance(self.env.action_space, gym.spaces.Discrete):
+            action_dim = self.env.action_space.n
+            disc = True
+        elif isinstance(self.env.action_space, gym.spaces.Box):
+            action_dim = self.env.action_space.shape[0]
+            disc = False
+        else:
+            raise NotImplementedError
+
+        self.q1 = get_model("v", self.network_type)(
+            state_dim, action_dim, "Qsa", self.layers
+        ).to(self.device)
+        self.q2 = get_model("v", self.network_type)(
+            state_dim, action_dim, "Qsa", self.layers
+        ).to(self.device)
+        self.policy = get_model("p", self.network_type)(
+            state_dim, action_dim, self.layers, disc, False, sac=True
+        )
+
+        if self.pretrained:
+            self.checkpoint = self.load(self.save_name, self.save_version)
```

Comment:
LGTM alert over here I think
It should just be `self.load(self)`
It does not return anything either directly sets a value to self.checkpoint